File: ios/Classes/BluetoothInfoPlugin.swift
import Flutter
import UIKit
import CoreBluetooth

public class SwiftBluetoothInfoPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "bluetooth_info", binaryMessenger: registrar.messenger())
    let instance = SwiftBluetoothInfoPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getDeviceName":
        let deviceName = getBluetoothDeviceName()
        result(deviceName)
    case "getDeviceAddress":
        let deviceAddress = getBluetoothDeviceAddress()
        result(deviceAddress)
    case "isBluetoothEnabled":
        let isBluetoothEnabled = isBluetoothEnabled()
        result(isBluetoothEnabled)
    default:
        result(FlutterMethodNotImplemented)
    }
  }

  private func getBluetoothDeviceName() -> String {
    let centralManager = CBCentralManager()
    return centralManager.identifier.uuidString
  }

  private func getBluetoothDeviceAddress() -> String {
    let centralManager = CBCentralManager()
    return centralManager.identifier.uuidString
  }

  private func isBluetoothEnabled() -> Bool {
    let centralManager = CBCentralManager()
    return centralManager.state == .poweredOn
  }
}

File: android/src/main/kotlin/com/example/bluetooth_info/BluetoothInfoPlugin.kt
package com.example.bluetooth_info

import android.bluetooth.BluetoothAdapter
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class BluetoothInfoPlugin : FlutterPlugin, MethodChannel.MethodCallHandler {

  private lateinit var channel: MethodChannel
  private lateinit var bluetoothAdapter: BluetoothAdapter

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(binding.binaryMessenger, "bluetooth_info")
    channel.setMethodCallHandler(this)
    bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
  }

  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
    when (call.method) {
      "getDeviceName" -> {
        val deviceName = getBluetoothDeviceName()
        result.success(deviceName)
      }
      "getDeviceAddress" -> {
        val deviceAddress = getBluetoothDeviceAddress()
        result.success(deviceAddress)
      }
      "isBluetoothEnabled" -> {
        val isBluetoothEnabled = isBluetoothEnabled()
        result.success(isBluetoothEnabled)
      }
      else -> {
        result.notImplemented()
      }
    }
  }

  private fun getBluetoothDeviceName(): String {
    return bluetoothAdapter.name ?: ""
  }

  private fun getBluetoothDeviceAddress(): String {
    return bluetoothAdapter.address ?: ""
  }

  private fun isBluetoothEnabled(): Boolean {
    return bluetoothAdapter.isEnabled
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
